prody/proteins/interactions.py

```diff
@@ -3010,6 +3239,484 @@ def showSminaTermValues(data):
+            else:
+                break
+
+        for n in range(len(seq2)):
+            fp2.write("{0} ".format(seq2[n]))
+            fp6.write("{0}".format(seq1[m]))
+            m += 1
+
+        for o in range(m, len(seq1)):
+            fp2.write("{0} ".format(seq1[m]))
+
+        fp2.write("\n")
+        fp6.write("\n{0}\n\n\n".format(tmpseq[0]))
+
+        mod_count += 1
+        fp1.write("MODEL\t{0}\n".format(mod_count))
+        fp7.write("MODEL\t{0}\n".format(mod_count))
+
+        for line in file4:
+            if line.strip():
+                fp7.write("{0}\n".format(line.strip()))
+
+        for line in tmpcord:
+            fp1.write("{0}".format(line))
+
+        fp1.write("TER\nENDMDL\n\n")
+        fp7.write("TER\nENDMDL\n\n")
+
+    # Cleanup
+    fp1.close()
+    fp2.close()
+    fp6.close()
+    fp7.close()
+    fp9.close()
+    fp10.close()
+
+    extractMultiModelPDB('aligned_structures.pdb', folder_name='struc_homologs')
+    subprocess.run("rm -f inp.pdb prot.seq target.pdb temp_coord.txt temp_resind.txt temp_seq.txt", shell=True)
+    subprocess.run("rm -rf tmp2 temp", shell=True)
+    
+
+def calcSignatureInteractions(mapping_file, PDB_folder, fixer='pdbfixer'):
+    """Analyzes protein structures to identify various interactions using InSty. 
+    Processes data from the MSA file and folder with selected models.
+    
+    :arg mapping_file: Aligned residue indices, MSA file type
+    :type mapping_file: str
+
+    :arg PDB_folder: Directory containing PDB model files
+    :type PDB_folder: str
+
+    :arg fixer: The method for fixing lack of hydrogen bonds
+    :type fixer: 'pdbfixer' or 'openbabel'
+    """
+    
+    import os
+    
+    functions = {
+        "HydrogenBonds": calcHydrogenBonds,
+        "SaltBridges": calcSaltBridges,
+        "RepulsiveIonicBonding": calcRepulsiveIonicBonding,
+        "PiStacking": calcPiStacking,
+        "PiCation": calcPiCation,
+        "Hydrophobic": calcHydrophobic,
+        "DisulfideBonds": calcDisulfideBonds
+    }
+
+    # Process each bond type
+    for bond_type, func in functions.items():
+        # Check if the consensus file already exists
+        consensus_file = '{}_consensus.txt'.format(bond_type)
+        if os.path.exists(consensus_file):
+            log_message("Consensus file for {} already exists, skipping.".format(bond_type), "INFO")
+            continue
+
+        log_message("Processing {}".format(bond_type))
+        result = process_data(mapping_file, PDB_folder, func, bond_type, fixer)
+
+        # Check if the result is None (no valid bonds found)
+        if result is None:
+            log_message("No valid {} entries found, skipping further processing.".format(bond_type), "WARNING")
+            continue
+
+        result, fixed_files = result
+
+        # Proceed with plotting
+        n = None
+        plot_barh(result, bond_type, n)
+
+    # Remove all fixed files at the end
+    if 'fixed_files' in locals():
```

Comment:
this is pointless. fixed_files was assigned above so it's always true unless something failed.

We shouldn't delete files based on whether a process finishes, but rather give the users the option with a kwarg

Reply:
I removed that line and added kwargs to determine whether we want to remove the temporary files (remove_tmp_files) or not. I am testing it right now (it is a time-consuming process because I need to wait till the very end).